src/dir.rs

```diff
@@ -237,10 +237,10 @@ impl<'a, T: ReadWriteSeek + 'a> Dir<'a, T> {
                 let entry = self.write_entry(name, sfn_entry)?;
                 let dir = entry.to_dir();
                 // create special entries "." and ".."
-                let dot_sfn = ShortNameGenerator::new(".").generate().unwrap();
+                let dot_sfn = ShortNameGenerator::dot().generate().unwrap();
```

Comment:
I think this is over-engineered (well, it's probably main fault). We don't need ShortNameGenerator for "fixed" names. We need just an 11-byte array here which you create in gen_fixed_entry() function but you also had to initialize all ShortNameGenerator fields. Please refactor it to simple methods returning those bytearrays e.g. generate_dot, generate_dot_dot

Reply:
Also please add regression tests checking if short name for those entries is properly set.

Reply:
You are right. I updated the PR.
Just a question for the regression tests:
The way we found it is reparsing the output with FatFs (the C version http://elm-chan.org/fsw/ff/00index_e.html)
And we found the special short entries `"           "` and `"~1        "` in hexdump.
Can you hint us a simple way to add this to the current tests?

Reply:
You can use DirEntry::short_file_name method to get the short name. It should be possible to test it in integration tests (in tests directory of repository)